File: ios/UUIDModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE(UUIDModule, UUIDModule, NSObject)

// UUID generation is sync and runs on the JS thread so in order to avoid performance
// issues, it should be as fast as possible.
RCT_EXTERN__BLOCKING_SYNCHRONOUS_METHOD(generate)

@end

File: ios/UUIDModule.swift
/**
 Native module for easy and fast unique ID generation on JS side. Used to generate native instance IDs.
 */
@objc(UUIDModule)
class UUIDModule: NSObject, RCTBridgeModule {
    /// Initialize this module on main thread.
    static func requiresMainQueueSetup() -> Bool {
        true
    }
    
    /// Exported JS module name.
    static func moduleName() -> String! {
        "UUIDModule"
    }
    
    /**
     Synchronously generate a random UUIDv4.
     - Returns: Random UUID RFC 4122 version 4.
     */
    @objc func generate() -> String {
        UUID().uuidString
    }
}
